File: BlackBerry/AlertMsgs/LockAlert.java
package com.mlhsoftware.Blurts;

import org.json.me.*;


class LockAlert extends AlertBase 
{
  // Tag Names
  private static final String ALERT_NAME = "Lock";
  private static final String KEY_TEXT = "Text";


  public LockAlert( String string ) throws JSONException
  {
    super( string );
  }

  public LockAlert()
  {
    super( TYPE_LOCK, ALERT_NAME );
  }

  public void setText( String value )
  {
    try
    {
      put( KEY_TEXT, value );
    }
    catch ( JSONException e )
    {
      System.out.println( e.toString() );
    }
  }
}

File: Desktop/Domain/AlertMsgs/LockAlert.cs
using System;
using System.Runtime.InteropServices;
using Blurts.Domain;

namespace Blurts
{
  [ComVisible(true)]
  public class LockAlert : AlertBase
  {
    // Tag Names
    private static String ALERT_NAME = "Lock";
    private static String KEY_STATUS = "Text";


    public LockAlert(String s)
      : base(s)
    {
    }

    public LockAlert()
      : base(DisplayItemType.STATUS, ALERT_NAME)
    {
    }

    [ComVisible(true)]
    public String Text
    {
      get
      {
        return optString(KEY_STATUS);
      }
      set
      {
        try
        {
          remove(KEY_STATUS);
          put(KEY_STATUS, value);
        }
        catch (JSONException e)
        {
          Console.WriteLine(e.ToString());
        }
      }
    }

    public override Boolean ProcessMessage()
    {
      if (ApplicationSettings.Instance.Active)
      {
        MainScreen.LockWorkStation();  //FIX move this call out of MainScreen
      }
      runScript( "OnLockAlert" );
      AlertHistory.Instance.AddAlert(XML);
      return true;
    }
  }
}